storage/integration_test.go

```diff
@@ -2251,41 +2251,53 @@ func TestIntegration_WriterContentType(t *testing.T) {
 }
 
 func TestIntegration_WriterChunksize(t *testing.T) {
-	ctx := skipJSONReads(skipGRPC("https://github.com/googleapis/google-cloud-go/issues/7798"), "no reads in test")
+	ctx := skipJSONReads(skipGRPC("https://github.com/googleapis/google-cloud-go/issues/7839"), "no reads in test")
 	multiTransportTest(ctx, t, func(t *testing.T, ctx context.Context, bucket, _ string, client *Client) {
 		obj := client.Bucket(bucket).Object("writer-chunksize-test" + uidSpaceObjects.New())
-		objSize := 17 << 10 << 10 // 17 Mib
+		objSize := 1<<10<<10 + 1 // 1 Mib + 1 byte
 		contents := bytes.Repeat([]byte("a"), objSize)
 
 		for _, test := range []struct {
 			desc             string
 			chunksize        int
 			wantBytesPerCall int64
+			wantCallbacks    int
 		}{
 			{
 				desc:             "default chunksize",
 				chunksize:        16 << 10 << 10,
 				wantBytesPerCall: 16 << 10 << 10,
+				wantCallbacks:    0,
 			},
 			{
 				desc:             "small chunksize rounds up to 256kib",
 				chunksize:        1,
 				wantBytesPerCall: 256 << 10,
+				wantCallbacks:    int(math.Ceil(float64(objSize) / (256 << 10))),
```

Comment:
This is confusing and opaque, just put the literal number of expected calls here I think.